component/loki/source/http/http.go

```diff
@@ -0,0 +1,170 @@
+package http
+
+import (
+	"context"
+	"reflect"
+	"sync"
+
+	"github.com/grafana/agent/pkg/util"
+
+	"github.com/go-kit/log/level"
+
+	"github.com/efficientgo/core/errors"
+	"github.com/grafana/agent/component"
+	"github.com/grafana/agent/component/common/loki"
+	"github.com/grafana/agent/component/common/relabel"
+	"github.com/grafana/agent/component/loki/source/http/internal/lokipush"
+	"github.com/prometheus/common/model"
+	"github.com/weaveworks/common/logging"
+	"github.com/weaveworks/common/server"
+)
+
+// TODO: this component also supports GRPC, so we may want to call it `loki.source.push_api` or something else.
+const componentName = "loki.source.http"
+
+type Arguments struct {
+	HTTPAddress          string              `river:"http_address,attr"`
+	HTTPPort             int                 `river:"http_port,attr"`
```

Comment:
We can default these to `0.0.0.0` and `3500` to match upstream defaults. I'm on the fence about making these required or optional.

These fields will be superseded by #3581 either way so there's no point nitpicking them too much at this stage, let's leave it as-is.

Check how other components make use of a `DefaultArguments` variable and an `UnmarshalRiver` method to enforce defaults and perform any validations.